### PR TITLE
NXPY-266: Align on user and group generated UID

### PR DESCRIPTION
**Pull request by gitofanindya**
## Summary by Sourcery

Introduce a username↔generated-UID translation layer in the Nuxeo Python client so requests send generated UIDs while responses expose usernames transparently.

Enhancements:
- Add client-side caching and helper methods to resolve usernames to generated UIDs and back, including support for prefixed values and bulk lists.
- Translate known username-carrying parameters in both URL query strings and automation operation bodies to generated UIDs before sending requests.
- Wrap HTTP responses so JSON decoding automatically converts generated UIDs back to usernames for known response fields without changing callers.
- Wire request/response translation into the operations API execution flow while keeping non-username parameters untouched.

Tests:
- Add an extensive unit test suite covering UID mapping, request/response translation, NuxeoClient.request integration, and API.execute integration, including round-trip and edge cases.

### CODE REVIEW

File: tests/unit/test_client_userid_mapper.py [hunk 1]
```diff
@@ -0,0 +1,751 @@
...
+"""
+from unittest.mock import MagicMock, patch
+
+import requests
+
+from nuxeo.client import NuxeoClient
+from nuxeo.operations import API
+
+# We do not need to set-up a server and log the current test
+skip_logging = True
```
[the remark this answers is not in the document]
Comment (gitofanindya) [reply tests/unit/test_client_userid_mapper.py:19]: done.

**Review by S-Ghosh-Dev2 — APPROVED** [review]
